eland/dataframe.py

```diff
@@ -1430,6 +1431,71 @@ def aggregate(
 
     hist = gfx.ed_hist_frame
 
+    def groupby(
+        self, by: Union[List[str], Dict[str, str], None] = None, dropna: bool = True
+    ) -> "GroupByDataFrame":
```

Comment:
Why is `Dict[str, str]` a type allowed by `by`?
Also since `by` is a required parameter we don't have to initiate it with `= None` do we? Or does pandas do this?

Reply:
Ah! I wanted to implement this: 
`ed_df.groupby("Cancelled").agg({"DestCountry": "mean", "AvgTicketPrice": "min"},True)`
But couldn't. Should remove that.
Maybe Ill do another PR for remaining things to be done in groupby